zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -36,6 +36,104 @@ sealed trait Server[-R, +E] { self =>
 
   def start(implicit ev: E <:< Throwable): ZIO[R with EventLoopGroup with ServerChannelFactory, Throwable, Nothing] =
     make.useForever
+
+  /**
+   * Launches the app with current settings: default EventLoopGroup (nThreads = 0) and ServerChannelFactory.auto.
+   */
+  def startDefault[R1 <: Has[_] with R](implicit ev: E <:< Throwable): ZIO[R1, Throwable, Nothing] =
+    start.provideSomeLayer[R1](EventLoopGroup.auto(0) ++ ServerChannelFactory.auto)
+
+  /**
+   * Creates a new server with the maximum size of the request specified in bytes.
+   */
+  def withMaxRequestSize(size: Int): Server[R, E] = Concat(self, Server.MaxRequestSize(size))
+
+  /**
+   * Creates a new server listening on the provided port.
+   */
+  def withPort(port: Int): Server[R, E] = Concat(self, Server.Address(new InetSocketAddress(port)))
+
+  /**
+   * Creates a new server listening on the provided hostname and port.
+   */
+  def withBind(hostname: String, port: Int): Server[R, E] =
+    Concat(self, Server.Address(new InetSocketAddress(hostname, port)))
+
+  /**
+   * Creates a new server listening on the provided InetAddress and port.
+   */
+  def withBind(address: InetAddress, port: Int): Server[R, E] =
+    Concat(self, Server.Address(new InetSocketAddress(address, port)))
+
+  /**
+   * Creates a new server listening on the provided InetSocketAddress.
+   */
+  def withBind(inetSocketAddress: InetSocketAddress): Server[R, E] = Concat(self, Server.Address(inetSocketAddress))
+
+  /**
+   * Creates a new server with the errorHandler provided.
+   */
+  def withError[R1](errorHandler: Throwable => ZIO[R1, Nothing, Unit]): Server[R with R1, E] =
+    Concat(self, Server.Error(errorHandler))
+
+  /**
+   * Creates a new server with the following ssl options.
+   */
+  def withSsl(sslOptions: ServerSSLOptions): Server[R, E] = Concat(self, Server.Ssl(sslOptions))
+
+  /**
+   * Creates a new server using a HttpServerExpectContinueHandler to send a 100 HttpResponse if necessary.
+   */
+  def withAcceptContinue: Server[R, E] = Concat(self, Server.AcceptContinue)
+
+  /**
+   * Creates a new server using netty FlowControlHandler (@see <a
+   * href="https://netty.io/4.1/api/io/netty/handler/flow/FlowControlHandler.html">FlowControlHandler</a>).
+   */
+  def withFlowControl: Server[R, E] = Concat(self, Server.FlowControl)
```

Comment:
May be we can use —

```scala
def withFlowControl(enable: Boolean) = ???
```

Reply:
Changed several methods to use an enable parameter